var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -371,3 +371,41 @@ def check_install(self):
                 print('-' * 80)
                 raise RuntimeError("HDF5 install check failed")
         shutil.rmtree(checkdir)
+
+    def _test_check_versions(self):
+        """Perform version checks on selected installed package binaries."""
+        spec_vers_str = 'Version {0}'.format(self.spec.version)
+        spec_mm = str(self.spec.version.up_to(2))
+
+        checks = {
+            'h52gif':  ([spec_vers_str], None),
+            'h5copy':  ([spec_vers_str], None),
+            'h5diff':  ([spec_vers_str], None),
+            'h5dump':  ([spec_vers_str], None),
+            'h5format_convert':  ([spec_vers_str], None),
+            'h5ls':  ([spec_vers_str], None),
+            'h5mkgrp':  ([spec_vers_str], None),
+            'h5perf':  ([spec_vers_str], 1),
+            'h5perf_serial':  ([spec_vers_str], 1),
+            'h5redeploy':  (['Unknown Option'], 1),
+            'h5repack':  ([spec_vers_str], None),
+            'h5repart':  ([spec_mm], None),
+            'h5stat':  ([spec_vers_str], None),
+            'h5unjam':  ([spec_vers_str], None),
```

Comment:
This is overkill, let's choose some reasonable set of "canonical" hdf5 executables.

Reply:
IIRC these were common across all the versions I ran against.

Reply:
Then let's just pull out the ones that don't respond to the version argument by printing it and exiting zero, so we can clean this up to just be a list of things to test.

For example, if they fixed any executable to properly respond to `--version` I don't want to report that as a failure in a future hdf5 version.